tests/src/androidTest/java/com/virgilsecurity/android/ethree/java/interaction/EThreeScopesTest.java

```diff
@@ -0,0 +1,157 @@
+ *  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package com.virgilsecurity.android.ethree.java.interaction;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import com.virgilsecurity.android.ethree.kotlin.callback.OnCompleteListener;
+import com.virgilsecurity.android.ethree.kotlin.callback.OnGetTokenCallback;
+import com.virgilsecurity.android.ethree.kotlin.callback.OnResultListener;
+import com.virgilsecurity.android.ethree.kotlin.interaction.EThree;
+import com.virgilsecurity.android.ethree.utils.TestConfig;
+import com.virgilsecurity.sdk.cards.Card;
+import com.virgilsecurity.sdk.cards.CardManager;
+import com.virgilsecurity.sdk.cards.validation.VirgilCardVerifier;
+import com.virgilsecurity.sdk.client.VirgilCardClient;
+import com.virgilsecurity.sdk.client.exceptions.VirgilServiceException;
+import com.virgilsecurity.sdk.common.TimeSpan;
+import com.virgilsecurity.sdk.crypto.VirgilAccessTokenSigner;
+import com.virgilsecurity.sdk.crypto.VirgilCardCrypto;
+import com.virgilsecurity.sdk.crypto.exceptions.CryptoException;
+import com.virgilsecurity.sdk.jwt.JwtGenerator;
+import com.virgilsecurity.sdk.jwt.accessProviders.GeneratorJwtProvider;
+import com.virgilsecurity.sdk.storage.DefaultKeyStorage;
+import com.virgilsecurity.sdk.storage.KeyStorage;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import kotlin.coroutines.CoroutineContext;
+import kotlinx.coroutines.CoroutineScope;
+import kotlinx.coroutines.Dispatchers;
+import kotlinx.coroutines.Job;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * EThreeScopesTest
+ */
+@RunWith(AndroidJUnit4.class)
+public class EThreeScopesTest {
+
+    private static final int NUMBER_OF_REPEATS = 10;
+    private static final long A_FEW_NETWORK_CALLS_DELAY = 1500L;
+
+    private String identity = UUID.randomUUID().toString();
+    private String tokenString;
+    private EThree eThree;
+    private KeyStorage keyStorage;
+
+    private JwtGenerator jwtGenerator;
+
+    @Before
+    public void setup() {
+        jwtGenerator = new JwtGenerator(TestConfig.Companion.getAppId(),
+                                        TestConfig.Companion.getApiKey(),
+                                        TestConfig.Companion.getApiPublicKeyId(),
+                                        TimeSpan.fromTime(600, TimeUnit.SECONDS),
+                                        new VirgilAccessTokenSigner(TestConfig.Companion.getVirgilCrypto()));
+
+        try {
+            tokenString = jwtGenerator.generateToken(identity).stringRepresentation();
+        } catch (CryptoException e) {
+            fail(e.getMessage());
```

Comment:
I'm not sure if 'fail' works at @Before method

Reply:
yep, tested - it works